src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -944,9 +1037,34 @@ public FormValidation doCheckUseInstanceProfileForCredentials(@QueryParameter bo
             return FormValidation.ok();
         }
 
-        public FormValidation doCheckPrivateKey(@QueryParameter String value) throws IOException, ServletException {
+        public ListBoxModel doFillSshKeysCredentialsIdItems(@QueryParameter String sshKeysCredentialsId) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+            StandardListBoxModel result = new StandardListBoxModel();
+
+            return result
+                    .includeMatchingAs(Jenkins.getAuthentication(), Jenkins.get(), BasicSSHUserPrivateKey.class, Collections.<DomainRequirement>emptyList(), CredentialsMatchers.always())
+                    .includeMatchingAs(ACL.SYSTEM, Jenkins.get(), BasicSSHUserPrivateKey.class, Collections.<DomainRequirement>emptyList(), CredentialsMatchers.always())
+                    .includeCurrentValue(sshKeysCredentialsId);
+        }
+
+        public FormValidation doCheckSshKeysCredentialsId(@QueryParameter String value) throws IOException, ServletException {
```

Comment:
I'm not 100% certain but it might need to `@RequirePOST`

Reply:
No other doCheckX method has this annotation in place. How to find out?

Reply:
@jvz @MRamonLeon do either of you know if this is required?

Reply:
There was a ping on twitter regarding this Pr and jira issue this is addressing so I thought I would jump in to help. A @RequirePOST is required when doing `doCheckX`
@djesionek

Reply:
I would say no, as long as the method is not modifying any state. Maybe @daniel-beck can confirm.

Reply:
https://www.jenkins.io/doc/developer/security/form-validation/ explains it.

At a glance, depends on exactly what `getSshCredential` does. If there are potentially network lookups for credentials involved, it's safer to `@RequirePOST` (or `@POST`).

Reply:
It only calls the `CredentialsProvider`, so `@RequirePOST` is not necessary in that case?

Reply:
Unsure how expensive network lookups can become with external credentials providers.

I mean, when in doubt add it; it's a reasonable default.

Reply:
Makes sense, didn't thought about external credentials providers at all. So I will add it. Thanks :+1: